jfix-zookeeper/src/main/kotlin/ru/fix/zookeeper/lock/ActiveLocksContainer.kt

```diff
@@ -0,0 +1,117 @@
+package ru.fix.zookeeper.lock
+
+import mu.KotlinLogging
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.ConcurrentMap
+import java.util.concurrent.locks.ReentrantLock
+
+class ActiveLocksContainer : AutoCloseable {
+    companion object {
+        private val logger = KotlinLogging.logger { }
+    }
+
+    /**
+     * remove, prolong and getState operations MUST BE synced under [.globalLock]
+     */
+    private val locks: ConcurrentMap<LockIdentity, LockContainer> = ConcurrentHashMap()
+
+    /**
+     * all operations with accessing to [.locks] has been synced with this object
+     * LockContainer is encapsulated inside this class so for performance issue this lock could be
+     * replaced with syncing via LockContainer
+     */
+    private val globalLock = ReentrantLock(true)
+
+    /**
+     * @return - old value in the map
+     */
+    fun putLock(
+            lockId: LockIdentity,
+            lock: PersistentExpiringDistributedLock,
+            listener: LockProlongationFailedListener
+    ): PersistentExpiringDistributedLock? {
+        val newLockContainer = LockContainer(lock, listener)
+        val oldLockContainer = locks.put(lockId, newLockContainer)
+        return oldLockContainer?.lock
+    }
+
+    fun removeLock(lockId: LockIdentity): PersistentExpiringDistributedLock? {
+        return try {
+            globalLock.lock()
```

Comment:
I actually don't see the point in locking the whole container here. How the locking of ConcurrentHashMap can help us?